armadillo/src/main/java/org/molgenis/armadillo/controller/StorageController.java

```diff
@@ -0,0 +1,292 @@
+import static org.springframework.web.bind.annotation.RequestMethod.HEAD;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.List;
+import java.util.Map;
+import javax.validation.Valid;
+import org.molgenis.armadillo.audit.AuditEventPublisher;
+import org.molgenis.armadillo.exceptions.FileProcessingException;
+import org.molgenis.armadillo.storage.ArmadilloStorageService;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@Tag(name = "storage", description = "API to manipulate the storage")
+@RestController
+@Valid
+@SecurityRequirement(name = "http")
+@SecurityRequirement(name = "bearerAuth")
+@SecurityRequirement(name = "JSESSIONID")
+@RequestMapping("storage")
+public class StorageController {
+
+  private final ArmadilloStorageService storage;
+  private final AuditEventPublisher auditor;
+
+  public StorageController(ArmadilloStorageService storage, AuditEventPublisher auditor) {
+    this.storage = storage;
+    this.auditor = auditor;
+  }
+
+  @Operation(summary = "List all projects")
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "Projects listed",
+            content =
+                @Content(array = @ArraySchema(schema = @Schema(implementation = String.class)))),
+        @ApiResponse(
+            responseCode = "401",
+            description = "Unauthorized",
+            content = @Content(schema = @Schema(hidden = true)))
+      })
+  @GetMapping(
+      value = "/projects",
+      produces = {APPLICATION_JSON_VALUE})
+  @ResponseStatus(OK)
+  public List<String> listProjects(Principal principal) {
+    return auditor.audit(storage::listProjects, principal, LIST_PROJECTS, Map.of());
+  }
+
+  @Operation(summary = "Project exists?")
```

Comment:
what is use case of this 'head' on projects? Seems to be unneeded?

Reply:
Btw, I get impression your quality of annotation is much better then what I did before on AdminController. I can imagine you would want to also put some TLC in the other controllers to make them just as well documented?

Reply:
More out of completeness I think, I thought it might come in handy in the client. But since we raise a `404` when the project doesn't exist, it's not really needed

Reply:
Yes I had fun exploring the annotations a bit 😄  I agree it would be nice to have such extensive documentation on other APIs as well, I'll take a look